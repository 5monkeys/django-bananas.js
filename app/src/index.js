/* eslint-disable no-unused-vars */
import Badge from "@material-ui/core/Badge";
import { blue } from "@material-ui/core/colors";
import AppleIcon from "@material-ui/icons/ColorLens";
import DashboardIcon from "@material-ui/icons/Dashboard";
import FaceIcon from "@material-ui/icons/Face";
import EmotIcon from "@material-ui/icons/InsertEmoticon";
import MailIcon from "@material-ui/icons/Mail";
import PeopleIcon from "@material-ui/icons/People";
import SettingsIcon from "@material-ui/icons/Settings";
import Bananas from "django-bananas";
import { django } from "django-bananas/colors";
import themes from "django-bananas/themes";
import React from "react";
import ReactDOM from "react-dom";

import settings from "./bananas.settings";
import Gravatar from "./Gravatar";

const exampleAppTheme = themes.default.extend({
  palette: {
    primary: {
      // main: blue[500],
    },
  },
  overrides: {
    BananasMenuItem: {
      avatar: {
        overflow: "visible",
      },
    },
    BananasNavBar: {
      header: {
        /* Example: Change NavBar header color */
        //   background: "#181818",
        /* Example: Make NavBar header striped */
        //   backgroundColor: django[700],
        //   background: "repeating-linear-gradient( 45deg, rgba(0, 0, 0, 0.12), rgba(0, 0, 0, 0.12) 10px, transparent 10px, transparent 20px)",
      },
      drawer: {
        /* Example: Change drawer width */
        //   width: 350,
      },
    },
    BananasTitleBar: {
      colorPrimary: {
        /* Example: Change primary colored TitleBar's */
        //   background: blue[500],
      },
    },
    /*
    BananasContainer: {
      root: {
        // Example: Center components using the Container
        maxWidth: 900,
        margin: "0 auto",
        width: "100%",
      },
    },
    */
  },
});

/*
const examplePageTheme = exampleAppTheme.extend({
  palette: {
    primary: {
      main: "#ff5500",
      dark: "#993300",
    },
  },
  overrides: {
    MuiButton: {
      outlined: {
        textTransform: "none",
      },
    },
  },
});
*/

const CustomIconComponent = () => {
  return (
    <Badge color="secondary" badgeContent={4} invisible={false}>
      <MailIcon />
    </Badge>
  );
};

const CustomLoginForm = () => <form>This is a custom login form</form>;

ReactDOM.render(
  <Bananas.App
    pages={route => import(`./pages/${route}`)}
    logLevel="DEBUG"
    layout="vertical" // horizontal|vertical
    title="Example"
    theme={exampleAppTheme}
    // pageTheme={examplePageTheme}
    // loginForm={CustomLoginForm}
    // branding="Admin"
    // version="v0.1"
    // logo={true} // true|URL|node
    // permanent={true}
    // collapsed: false,
    // dense: true,
    nav={{
      "fruit.peach:list": { icon: CustomIconComponent, title: "Custom Peach" },
      "fruit.apple:red_delicious": { icon: FaceIcon, title: "Red apple" },
      "fruit.apple:granny_smith": FaceIcon,
      "fruit.apple:list": AppleIcon,
      "bananas.me:list": Gravatar,
      // home: DashboardIcon,
      // "bananas.me:list": SettingsIcon,
      "fruit.banana:list": { parent: "peach", icon: PeopleIcon },
      "example.user:list": PeopleIcon,
    }}
    editableSettings
    {...settings}
  />,
  document.getElementById("root")
);
