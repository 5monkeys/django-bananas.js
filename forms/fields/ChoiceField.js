"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const ChoiceField = (_ref) => {
  let input = _ref.input,
      schema = _ref.schema,
      fieldProps = _ref.fieldProps;
  return _react.default.createElement(_core.FormControl, {
    error: fieldProps.error
  }, _react.default.createElement(_core.InputLabel, {
    htmlFor: "name-error"
  }, fieldProps.label), _react.default.createElement(_core.Select, _extends({}, input, {
    value: input.value || []
  }), !schema.required && _react.default.createElement(_core.MenuItem, {
    value: ""
  }, _react.default.createElement("em", null, "None")), schema.enum.map(value => _react.default.createElement(_core.MenuItem, {
    value: value,
    key: value
  }, value))), fieldProps.error && _react.default.createElement(_core.FormHelperText, null, fieldProps.helperText));
};

var _default = ChoiceField;
exports.default = _default;