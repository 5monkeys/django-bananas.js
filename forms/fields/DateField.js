"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _pickers = require("@material-ui/pickers");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var DateField = (_ref) => {
  var {
    input: {
      onChange: _onChange
    },
    fieldProps
  } = _ref,
      inputProps = _objectWithoutProperties(_ref.input, ["onChange"]);

  return _react.default.createElement(_pickers.DatePicker, _extends({}, inputProps, fieldProps, {
    onChange: date => {
      _onChange(date.toISOString().split("T")[0]);
    }
  }));
};

var _default = DateField;
exports.default = _default;