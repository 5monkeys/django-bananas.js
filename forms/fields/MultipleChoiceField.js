"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var MultipleChoiceField = (_ref) => {
  var {
    input,
    schema,
    fieldProps
  } = _ref;
  return _react.default.createElement(_core.FormControl, {
    error: fieldProps.error
  }, _react.default.createElement(_core.InputLabel, {
    htmlFor: "name-error"
  }, fieldProps.label), _react.default.createElement(_core.Select, _extends({}, input, {
    value: input.value instanceof Array ? input.value : input.value ? [input.value] : [],
    multiple: true
  }), !schema.required && _react.default.createElement(_core.MenuItem, {
    value: ""
  }, _react.default.createElement("em", null, "None")), schema.items.enum.map(value => _react.default.createElement(_core.MenuItem, {
    value: value,
    key: value
  }, value))), fieldProps.error && _react.default.createElement(_core.FormHelperText, null, fieldProps.helperText));
};

var _default = MultipleChoiceField;
exports.default = _default;