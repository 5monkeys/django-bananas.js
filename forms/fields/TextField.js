"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@material-ui/core");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const TextField = (_ref) => {
  let input = _ref.input,
      fieldProps = _ref.fieldProps;
  return _react.default.createElement(_core.TextField, _extends({}, input, fieldProps));
};

var _default = TextField;
exports.default = _default;