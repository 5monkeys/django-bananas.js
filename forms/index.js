"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AutoField", {
  enumerable: true,
  get: function get() {
    return _AutoField.default;
  }
});
Object.defineProperty(exports, "FieldArray", {
  enumerable: true,
  get: function get() {
    return _FieldArray.default;
  }
});
Object.defineProperty(exports, "Form", {
  enumerable: true,
  get: function get() {
    return _Form.default;
  }
});

var _AutoField = _interopRequireDefault(require("./AutoField"));

var _FieldArray = _interopRequireDefault(require("./FieldArray"));

var _Form = _interopRequireDefault(require("./Form"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }