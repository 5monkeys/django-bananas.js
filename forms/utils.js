"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.fieldFromSchema = fieldFromSchema;
exports.fieldFromErrorResponse = fieldFromErrorResponse;

function normalizePath(path) {
  return path.replace("]", "").replace("[", ".");
} // TODO: more verbose error messages


function fieldFromSchema(schema, field) {
  var normalizedSchema = {
    type: "object",
    properties: schema
  };
  var normalizedPath = normalizePath(field);
  return normalizedPath.split(".").reduce((acc, key) => {
    if (acc.type === "object") {
      var value = acc.properties[key];

      if (typeof value === "undefined") {
        throw new Error("Encountered a non-existent key \"".concat(key, "\"."));
      }

      return value;
    } else if (acc.type === "array") {
      if (!key.match(/^\d+$/)) {
        throw new Error("Encountered a non-numeric index \"".concat(key, "\" access on an array."));
      }

      return acc.items;
    }

    throw new Error("Encountered lookup \"".concat(key, "\" on unsupported type \"").concat(acc.type, "\"."));
  }, normalizedSchema);
}

function fieldFromErrorResponse(response, field) {
  var normalizedPath = normalizePath(field);
  return normalizedPath.split(".").reduce((acc, key) => acc[key.match(/^\d+$/) ? Number(key) : key], response);
}