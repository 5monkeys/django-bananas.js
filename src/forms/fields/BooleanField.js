import {
  Checkbox,
  FormControl,
  FormControlLabel,
  FormGroup,
  FormHelperText,
  Switch,
} from "@material-ui/core";
import React from "react";

const BooleanField = ({
  input: { value, ...inputProps },
  variant,
  fieldProps,
}) => (
  <FormControl required error={fieldProps.error}>
    <FormGroup>
      <FormControlLabel
        control={
          variant === "switch" ? (
            <Switch checked={value} {...inputProps} />
          ) : (
            <Checkbox checked={value} {...inputProps} />
          )
        }
        {...(fieldProps || {})}
      />
    </FormGroup>
    {fieldProps.error && (
      <FormHelperText>{fieldProps.helperText}</FormHelperText>
    )}
  </FormControl>
);

export default BooleanField;
