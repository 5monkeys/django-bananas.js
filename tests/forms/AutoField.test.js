import { render } from "@testing-library/react";
import Logger from "js-logger";
import React from "react";

import { AutoField, Form } from "../../src/forms";
import BooleanField from "../../src/forms/fields/BooleanField";
import ChoiceField from "../../src/forms/fields/ChoiceField";
import DateField from "../../src/forms/fields/DateField";
import DateTimeField from "../../src/forms/fields/DateTimeField";
import MultipleChoiceField from "../../src/forms/fields/MultipleChoiceField";
import TextField from "../../src/forms/fields/TextField";
import getAPIClient from "../api.mock";
import { TestContext } from "./utils";

Logger.get("bananas").setLevel(Logger.OFF);

beforeEach(() => {
  // surpress errors we're triggering by choice
  jest.spyOn(console, "error").mockImplementation(() => {});
  // hide MUI-warnings
  jest.spyOn(console, "warn").mockImplementation(() => {});
});

class Boundary extends React.Component {
  state = {};

  static getDerivedStateFromError(error) {
    return { error };
  }

  render() {
    const { children } = this.props;
    const { error } = this.state;
    return error != null ? error.message : children;
  }
}

test("Get a friendly reminder about FormContext, if missing", () => {
  const { getByText } = render(
    <Boundary>
      <AutoField name="test" />
    </Boundary>
  );

  expect(
    getByText("This component requires FormContext to be present.")
  ).toBeTruthy();
});

test.each([
  ["boolean", "checkbox", BooleanField],
  ["boolean", "switch", BooleanField],
  ["choices", "default", ChoiceField],
  ["date", "default", DateField],
  ["datetime", "default", DateTimeField],
  ["integer", "default", TextField],
  ["multiple_choices", "default", MultipleChoiceField],
  ["text", "default", TextField],
])("Can render field of type '%s' and variant '%s'", async (name, variant) => {
  const api = await getAPIClient();

  const { getByTestId } = render(
    <TestContext api={api}>
      <Form route="example.user:form.create">
        <AutoField
          name={name}
          variant={variant}
          fieldProps={{ "data-testid": name }}
        />
      </Form>
    </TestContext>
  );
  expect(getByTestId(name)).toBeTruthy();
});

const CustomFieldsByType = {
  string: {
    default: {
      component: ({ fieldProps = {} }) => <input {...fieldProps} />,
    },
    date: {
      component: ({ fieldProps = {} }) => <input {...fieldProps} />,
      "date-time": {
        component: ({ fieldProps = {} }) => <input {...fieldProps} />,
      },
    },
    boolean: {
      default: {
        component: ({ fieldProps = {} }) => <input {...fieldProps} />,
        type: "checkbox",
      },
    },
    enum: {
      default: {
        component: ({ fieldProps = {} }) => <input {...fieldProps} />,
        type: "select",
      },
    },
    array: {
      default: {
        component: ({ fieldProps = {} }) => <input {...fieldProps} />,
        type: "select",
      },
    },
  },
};

test.each([
  ["boolean", "checkbox", "boolean"],
  ["boolean", "switch", "boolean"],
  ["date", "default", "date"],
  ["datetime", "default", "datetime"],
  ["integer", "default", "text"],
  ["multiple_choices", "default", "array"],
  ["text", "default", "text"],
])(
  "Can render field of type '%s' and variant '%s' with custom field mapping",
  async (name, variant) => {
    const api = await getAPIClient();
    const { getByTestId } = render(
      <TestContext api={api}>
        <Form route="example.user:form.create">
          <AutoField
            fieldProps={{ "data-testid": name }}
            fieldsByType={CustomFieldsByType}
            name={name}
            variant={variant}
          />
        </Form>
      </TestContext>
    );
    expect(getByTestId(name)).toBeTruthy();
  }
);

const DateTimeComponent = ({ fieldProps = {} }) => <input {...fieldProps} />;

const CustomPartialFieldsByType = {
  string: {
    "date-time": { component: DateTimeComponent },
  },
};

test.each([
  ["boolean", "checkbox", BooleanField],
  ["boolean", "switch", BooleanField],
  ["choices", "default", ChoiceField],
  ["date", "default", DateField],
  ["datetime", "default", DateTimeComponent],
  ["integer", "default", TextField],
  ["multiple_choices", "default", MultipleChoiceField],
  ["text", "default", TextField],
])(
  "Can render field of type '%s' and variant '%s' with partial fieldsByType",
  async (name, variant) => {
    const api = await getAPIClient();
    const { getByTestId } = render(
      <TestContext api={api}>
        <Form route="example.user:form.create">
          <AutoField
            fieldProps={{ "data-testid": name }}
            fieldsByType={CustomPartialFieldsByType}
            name={name}
            variant={variant}
          />
        </Form>
      </TestContext>
    );
    expect(getByTestId(name)).toBeTruthy();
  }
);

const modifiedTextField = props => (
  <div>
    <TextField {...props} fieldProps={{ "data-testid": "t1" }} />
  </div>
);

test("Can use FieldComponent prop to render a custom component", async () => {
  const api = await getAPIClient();
  const { getByTestId } = render(
    <Boundary>
      <TestContext api={api}>
        <Form route="example.user:form.create">
          <AutoField name="text" FieldComponent={modifiedTextField} />
        </Form>
      </TestContext>
    </Boundary>
  );
  expect(getByTestId("t1")).toBeTruthy();
});
