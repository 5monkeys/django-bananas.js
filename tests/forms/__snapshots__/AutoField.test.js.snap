// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can render field of type 'boolean' and variant 'checkbox' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root"
  >
    <div
      className="MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
            <input
              checked={false}
              className="PrivateSwitchBase-input-22"
              data-indeterminate={false}
              disabled={false}
              name="boolean"
              onChange={[Function]}
              type="checkbox"
            />
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Boolean *
        </span>
      </label>
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'boolean' and variant 'switch' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root"
  >
    <div
      className="MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          className="MuiSwitch-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiSwitch-thumb"
              />
              <input
                checked={false}
                className="PrivateSwitchBase-input-22 MuiSwitch-input"
                disabled={false}
                name="boolean"
                onChange={[Function]}
                type="checkbox"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Boolean *
        </span>
      </label>
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'choices' and variant 'default' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="name-error"
    >
      Choices *
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        id="select-choices"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        name="choices"
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'date' and variant 'default' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
    format="MMMM do"
    onClick={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
    >
      Date *
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="date"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="July 1st"
      />
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'datetime' and variant 'default' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
    format="MMMM do hh:mm aaaa"
    onClick={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
    >
      Datetime *
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="datetime"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
        value="July 1st 03:54 p.m."
      />
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'integer' and variant 'default' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Integer *
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="integer"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'multiple_choices' and variant 'default' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="name-error"
    >
      undefined *
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        id="select-multiple_choices"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        name="multiple_choices"
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</form>
`;

exports[`Can render field of type 'text' and variant 'default' 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Text *
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="text"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
